java/ql/src/Security/CWE/CWE-352/CsrfUnprotectedRequestType.qhelp

```diff
@@ -0,0 +1,72 @@
+<!DOCTYPE qhelp SYSTEM "qhelp.dtd">
+<qhelp>
+
+<overview>
+    <p>
+      Cross-site request forgery (CSRF) is a type of vulnerability in which an
+      attacker is able to force a user to carry out an action that the user did
+      not intend.
+    </p>
+
+    <p>
+      The attacker tricks an authenticated user into submitting a request to the
+      web application. Typically this request will result in a state change on
+      the server, such as changing the user's password. The request can be
+      initiated when the user visits a site controlled by the attacker. If the
+      web application relies only on cookies for authentication, or on other
+      credentials that are automatically included in the request, then this
+      request will appear as legitimate to the server.
+    </p>
+</overview>
+
+<recommendation>
+<p>Make sure any requests that change application state are protected from CSRF. Some application
+frameworks provide default CSRF protection for unsafe HTTP request methods (such as <code>POST</code>)
+which may change the state of the application. Safe HTTP request methods (such as <code>GET</code>)
+should only perform read-only operations and should not be used for actions that change application
+state.</p>
+
+<p>This query currently supports the Spring and Stapler web frameworks. Spring provides default CSRF protection
+for all unsafe HTTP methods whereas Stapler provides default CSRF protection for the <code>POST</code> method.</p>
+</recommendation>
+
+<example>
+<p> The following examples show Spring request handlers allowing safe HTTP request methods for state-changing actions.
+Since safe HTTP request methods do not have default CSRF protection in Spring, they should not be used when modifying
+application state. Instead, use one of the unsafe HTTP methods which Spring default-protects from CSRF.</p>
```

Comment:
This is me not understanding this properly (I am not a developer), but double-checking that we are recommending that users use UNSAFE HTTP methods, which despite being unsafe, seem to default-protect against CSRF. It is a bit confusing to me. Wondering if we should remove the word "unsafe" here 🤔 
Feel free to ignore this comment if the current content is accurate.

Reply:
> double-checking that we are recommending that users use UNSAFE HTTP methods

Yes, we're recommending using unsafe HTTP methods, which does sound confusing, but it is the correct terminology for the subset of HTTP methods that should be used, so I'll leave this as-is.